aten/src/ATen/native/MatrixExponential.cpp

```diff
@@ -0,0 +1,86 @@
+#include <ATen/native/MatrixExponential.h>
+
+namespace at { namespace native {
+
+DEFINE_DISPATCH(matrix_exp_stub);
+
+// Computes the matrix exponential for a given batch of squared matrices.
+// The implementaion is based on:
+//
+// Bader, P.; Blanes, S.; Casas, F.
+// Computing the Matrix Exponential with an Optimized Taylor Polynomial Approximation.
+// Mathematics 2019, 7, 1174.
+//
+Tensor matrix_exp(const Tensor& a) {
+  TORCH_CHECK(a.dim() >= 2 
+          && (at::isFloatingType(a.scalar_type()) 
+           || at::isComplexType(a.scalar_type())),
+              "matrix_exp(", a.scalar_type(), "{", a.sizes(), "}): expected a tensor "
+              "of floating types with dim at least 2");
+  TORCH_CHECK(a.size(-1) == a.size(-2),
+              "matrix_exp(", a.scalar_type(), "{", a.sizes(), "}): expected a tensor "
+              "of squared matrices");
+
+  // Case of 1x1 matrices
+  if (a.size(-1) == 1) {
+    return a.exp();
+  }
+
+  // Create input for kernels by squashing batch dimensions,
+  // i.e. make an n-dim tensor a 3-dim tensor
+  auto mexp = at::empty(a.sizes(), a.options());
+  auto mexp_squashed_batch_dims = mexp.view(
+    {-1, mexp.size(-2), mexp.size(-1)});
+  auto a_squashed_batch_dims = a.view(
+    {-1, a.size(-2), a.size(-1)});
```

Comment:
@ezyang , is it safe to use `view` in here? Or would `reshape` be better (in case of some crazy strides)? It is the first time I am dealing with the batched tensors, so I am not sure what is the proper way.

Reply:
Nevermind, turns out `matmul` does change the layout, so `reshape` will do.